File: scrabble-best-word/scrabbleBestWord.js
const scrabbleBestWord = (pointsArray, wordsArray) => {
    const alpha = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
    const pointsDict = {};
    pointsArray.forEach((element, index) => {
        pointsDict[alpha[index]] = element;
    });

    const values = wordsArray.map((element) => {
        let value = 0;
        element.split('').forEach((innerElement) => {
            value += pointsDict[innerElement];
        });
        return value + "." + element.length;
    });

    const equals = values.filter((element) => {
        if(Math.floor(element) === Math.floor(Math.max(...values))) {
            return element;
        };
    });

    return values.indexOf(Math.min(...equals).toString());
};

module.exports = { scrabbleBestWord };

//https://www.codewars.com/kata/563f960e3c73813942000015/train/javascript

File: python-venv/scrabble_best_word/kata_function/scrabble_best_word.py
import math

class Scrabble:
    def scrabble_best_world(pointsArray, wordsArray):
        alpha = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
        pointsDict = {}
        for i in range(len(pointsArray)):
            pointsDict[alpha[i]] = pointsArray[i]

        def find_value(word):
            value = 0
            for j in range(len(word)):
                value += pointsDict[word[j]]
            decimalValue = str(value) + "." + str(len(word))
            return float(decimalValue)

        def filter_value(number):
            if math.floor(number) == math.floor(max(values)):
                return number

        values = list(map(find_value, wordsArray))
        equals = list(filter(filter_value, values))

        return values.index((min(equals)))

#https://www.codewars.com/kata/563f960e3c73813942000015/train/python